app/src/main/java/an/maguste/android/navier/data/JsonLoad.kt

```diff
@@ -0,0 +1,107 @@
+package an.maguste.android.navier.data
+
+import android.content.Context
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+private val jsonFormat = Json { ignoreUnknownKeys = true }
+
+@Serializable
+private class JsonGenre(val id: Int, val name: String)
+
+@Serializable
+private class JsonActor(
+        val id: Int,
+        val name: String,
+        @SerialName("profile_path")
+        val profilePicture: String
+)
+
+@Serializable
+private class JsonMovie(
+        val id: Int,
+        val title: String,
+        @SerialName("poster_path")
+        val posterPicture: String,
+        @SerialName("backdrop_path")
+        val backdropPicture: String,
+        val runtime: Int,
+        @SerialName("genre_ids")
+        val genreIds: List<Int>,
+        val actors: List<Int>,
+        @SerialName("vote_average")
+        val ratings: Float,
+        val overview: String,
+        val adult: Boolean,
+        @SerialName("vote_count")
+        val reviews: Int
+)
+
+private suspend fun loadGenres(context: Context): List<Genre> = withContext(Dispatchers.IO) {
+    val data = readAssetFileToString(context, "genres.json")
+    parseGenres(data)
+}
+
+internal fun parseGenres(data: String): List<Genre> {
+    val jsonGenres = jsonFormat.decodeFromString<List<JsonGenre>>(data)
+    return jsonGenres.map { Genre(id = it.id, name = it.name) }
+}
+
+private fun readAssetFileToString(context: Context, fileName: String): String {
+    val stream = context.assets.open(fileName)
+
+    return stream.bufferedReader().readText()
```

Comment:
Открытые ресурсы (файлы, БД и прочее) нужно за собой закрывать. Для этих целей есть вспомогательная функция расширения `use` https://kotlinlang.org/docs/tutorials/kotlin-for-py/scoped-resource-usage.html
```suggestion
    return context.assets.open(fileName).use { stream ->
        stream.bufferedReader().readText()
    }
```